ui/src/app/pages/signed-in/nav-bar.tsx

```diff
@@ -64,235 +65,154 @@ const styles = reactStyles({
   }
 });
 
-export interface Props {
-  profile: Profile;
-  profileState: ProfileStore;
-  bannerAdminActive: boolean;
-  workspaceAdminActive: boolean;
-  headerImg: string;
-  displayTag: string;
-  shouldShowDisplayTag: boolean;
-  profileImage: string;
-  sidenavToggle: boolean;
-  homeActive: boolean;
-  workspacesActive: boolean;
-  libraryActive: boolean;
-  profileActive: boolean;
-  userAdminActive: boolean;
-  userAuditActive: boolean;
-  workspaceAuditActive: boolean;
-}
-
-export interface State {
-  sideNavVisible: boolean;
-  statusAlertVisible: boolean;
-  statusAlertDetails: {
-    statusAlertId: number;
-    title: string;
-    message: string;
-    link: string;
-  };
-  barsTransform: string;
-  hovering: boolean;
-  wrapperRef: React.RefObject<HTMLDivElement>;
-}
-
 const barsTransformNotRotated = 'rotate(0deg)';
 const barsTransformRotated = 'rotate(90deg)';
 
 const cookieKey = 'status-alert-banner-dismissed';
 
-export const NavBar = withUserProfile()(
-  class extends React.Component<Props, State> {
-    constructor(props) {
-      super(props);
-      // Bind the this context - this will be passed down into the actual
-      // sidenav / alert banner so clicks on it can close the modal
-      this.onToggleSideNav = this.onToggleSideNav.bind(this);
-      this.handleClickOutside = this.handleClickOutside.bind(this);
-      this.handleStatusAlertBannerUnmount = this.handleStatusAlertBannerUnmount.bind(this);
-      this.state = {
-        sideNavVisible: false,
-        statusAlertVisible: false,
-        statusAlertDetails: {
-          statusAlertId: 0,
-          title: '',
-          message: '',
-          link: '',
-        },
-        barsTransform: barsTransformNotRotated,
-        hovering: false,
-        wrapperRef: React.createRef(),
-      };
-    }
-
-    async componentDidMount() {
-      document.addEventListener('click', this.handleClickOutside);
+const shouldShowStatusAlert = (statusAlertId, statusAlertMessage) => {
+  if (cookiesEnabled()) {
+    const cookie = localStorage.getItem(cookieKey);
+    return (!cookie || (cookie && cookie !== `${statusAlertId}`)) && !!statusAlertMessage;
```

Comment:
Do we need a cookie not null check again after OR

Reply:
We've got one - that's what `cookie && cookie !== [...]` is doing

Reply:
I meant do we need one since we are checking !cookie first

Reply:
Oh, my misunderstanding - `!null` is valid Javascript, it'll return `true` because `null` has a truthiness value of `false`